EPOParser.py

```diff
@@ -4,115 +4,133 @@
 import datetime
 import functools
 
+
 def fmap(fn, coll):
-  return functools.reduce(lambda acc, val: acc + [fn(val)], coll, [])
+    return functools.reduce(lambda acc, val: acc + [fn(val)], coll, [])
+
 
 def ffilter(predicateFn, coll):
-  return functools.reduce(lambda acc, val: (acc + [val]) if predicateFn(val) else acc, coll, [])
+    return functools.reduce(
+        lambda acc, val: (acc + [val]) if predicateFn(val) else acc, coll, [])
+
 
 def fcmap(fn):
-  return lambda coll: functools.reduce(lambda acc, val: acc + [fn(val)], coll, [])
+    return lambda coll: functools.reduce(
+        lambda acc, val: acc + [fn(val)], coll, [])
+
 
 def fcfilter(predicateFn):
-  return lambda coll: functools.reduce(lambda acc, val: (acc + [val]) if predicateFn(val) else acc, coll, [])
+    return lambda coll: functools.reduce(
+        lambda acc, val: (acc + [val]) if predicateFn(val) else acc, coll, [])
+
 
 def fzip(coll1, coll2):
-  return list(zip(coll1, coll2))
+    return list(zip(coll1, coll2))
+
 
 def fcompose(*functions):
-  return functools.reduce(lambda f, g: lambda x: f(g(x)), functions, lambda x: x)  
+    return functools.reduce(
+        lambda f, g: lambda x: f(g(x)), functions, lambda x: x)
 
 ###########
-# helpers # 	
+# helpers #
 ###########
 
 def marker(id):
-  return id + '-parsecomplete'
+    return id + '-parsecomplete'
 
-# fromElement :: attrib | text -> Element -> {} | String
-def fromElement(info):
-  return lambda el: getattr(el, info)
+# from_element :: attrib | text -> Element -> {} | String
+def from_element(info):
+    """This is a docstring, which is used to comment on a function."""
+    return lambda el: getattr(el, info)
 
 # tree :: String -> Tree
 def tree(filename):
-  return etree.parse(filename, etree.XMLParser(encoding='utf-8', recover=True))
-
-# getFileInfo :: String -> [(filename, file-location)]
-def getFileInfo(indexFilename):
-  return fzip(
-    fmap(fromElement('text'), getElement(tree(indexFilename), '//filename')),
-    fmap(fromElement('text'), getElement(tree(indexFilename), '//file-location'))
-  )
-
-# getElement :: Tree, xpath -> [Element]
-def getElement(tree, elementPath):
-  return tree.xpath(elementPath)
-
-# getMetadata :: Tree, xpath, attrib | text, String -> [] | [(alias, tag, {} | String)]
-def getMetadata(tree, elementPath, info, alias):
-  return fmap(lambda el: ( alias, el.tag, getattr(el, info) ), getElement(tree, elementPath))
+    """
+    This is a multi-line docstring.
+
+    It is used when there is more info that should be shared than fits on one
+    line.
+    """
+    return etree.parse(
+        filename, etree.XMLParser(encoding='utf-8', recover=True))
+
+# get_file_info :: String -> [(filename, file-location)]
+def get_file_info(indexFilename):
+    return fzip(
```

Comment:
This also seems like a function where adding a docstring would make sense. That would give us an expectation about what we will find when we seek out these xpaths. (Basically something similar to your one-line comment, just formatted as a docstring. For example: "Returns a list of tuples, each containing the filename and path found in the dump's index file.")